scripts/restore.sh

```diff
@@ -67,6 +67,12 @@ if [ -f "$BACKUP_FILE" ]; then
             echo "RCON is not enabled. Please enable RCON to use this feature. Unable to restore backup."
             exit 1
         fi
+
+        while [ -n "$(pidof PalServer-Linux-Test)" ] 
+        do
+            echo "Waiting for Palworld to exit.."
+            sleep 1
+        done
```

Comment:
```suggestion
        mapfile -t server_pids < <(pgrep PalServer-Linux-Test)
        if [ "${#server_pids[@]}" -ne 0 ]; then
            echo "Waiting for Palworld to exit.."
            for pid in "${server_pids[@]}"; do
                tail --pid="$pid" -f 2>/dev/null
            done
        fi
```

I would remove the need for sleep using tail. 
Also I would use an array so in case there's multiple running.
Code is essentially what's in init.sh.